pkg/handlers/webhookreceiver.go

```diff
@@ -93,49 +100,91 @@ func (h *WebhookReceiverHandler) processAMReceiver(d AMReceiverData, ctx context
 		return &AMReceiverResponse{Error: err, Status: "unable to list managed notifications", Code: http.StatusInternalServerError}
 	}
 
+	// Handle each firing alert
 	for _, alert := range d.Alerts.Firing() {
-		// Ignore alerts which don't have a name
-		alertname, err := alertName(alert)
+		err = h.processAlert(alert, mnl)
 		if err != nil {
-			log.WithError(err).Error("alertname missing for alert")
-			continue
+			log.WithError(err).Error("an alert could not be successfully processed")
 		}
+	}
+	return &AMReceiverResponse{Error: nil, Status: "ok", Code: http.StatusOK}
+}
 
-		// Ignore alerts which don't have the send_managed_notification label
-		if val, ok := alert.Labels[AMLabelManagedNotification]; !ok || val == "false" {
-			log.WithField("alertname", alertname).Debug("ignoring alert with no send_managed_notification label")
-			continue
-		}
+// processAlert handles the pre-check verification and sending of a notification for a particular alert
+// and returns an error if that process completed successfully or false otherwise
+func (h *WebhookReceiverHandler) processAlert(alert template.Alert, mnl *oav1alpha1.ManagedNotificationList) error {
+	// Should this alert be handled?
+	if !isValidAlert(alert) {
+		log.WithField(LogFieldAlert, alert).Info("alert does not meet valid criteria")
+		return fmt.Errorf("alert does not meet valid criteria")
+	}
 
-		// Ignore alerts which don't have a notification defined
-		notificationName, foundNotification := alert.Labels[AMLabelTemplateName]
-		if !foundNotification {
-			log.WithField("alertname", alertname).Error("alert does not have template defined")
-			continue
-		}
+	// Can the alert be mapped to an existing notification definition?
+	notification, managedNotifications, err := getNotification(alert.Labels[AMLabelManagedNotification], mnl)
```

Comment:
```suggestion
	notification, managedNotifications, err := getNotification(alert.Labels[AMLabelTemplateName], mnl)
```
Testing this, came across below error and realised that the label is incorrect:
```
time="2022-03-15T11:55:31Z" level=error msg="an alert could not be successfully processed" error="matching managed notification not found for true"
```

After fixing the label, got below log:
```
time="2022-03-15T13:20:08Z" level=info msg="will send servicelog for notification" notification=TestTemplate
```